include/deal.II/lac/petsc_precondition.h

```diff
@@ -684,21 +684,58 @@ namespace PETScWrappers
      */
     struct AdditionalData
     {
+      /**
+       * Defines the available relax types for BoomerAMG.
+       */
+      enum class RelaxationType
+      {
+        Jacobi,
+        sequentialGaussSeidel,
+        seqboundaryGaussSeidel,
+        SORJacobi,
+        backwardSORJacobi,
+        symmetricSORJacobi,
+        l1scaledSORJacobi,
+        GaussianElimination,
+        l1GaussSeidel,
+        backwardl1GaussSeidel,
+        CG,
+        Chebyshev,
+        FCFJacobi,
+        l1scaledJacobi,
+        None
+      };
+
       /**
        * Constructor. Note that BoomerAMG offers a lot more options to set
        * than what is exposed here.
        */
-      AdditionalData(const bool         symmetric_operator = false,
-                     const double       strong_threshold   = 0.25,
-                     const double       max_row_sum        = 0.9,
-                     const unsigned int aggressive_coarsening_num_levels = 0,
-                     const bool         output_details = false);
+      AdditionalData(
+        const bool           symmetric_operator               = false,
+        const double         strong_threshold                 = 0.25,
+        const double         max_row_sum                      = 0.9,
+        const unsigned int   aggressive_coarsening_num_levels = 0,
+        const bool           output_details                   = false,
+        const RelaxationType relaxation_type_up   = RelaxationType::SORJacobi,
+        const RelaxationType relaxation_type_down = RelaxationType::SORJacobi,
+        const RelaxationType relaxation_type_coarse =
+          RelaxationType::GaussianElimination,
+        const unsigned int n_sweeps_coarse = 1,
+        const double       tol             = 0.0,
+        const unsigned int max_iter        = 1,
+        const bool         w_cycle         = false);
+
+      /**
+       * Converts the enums for the different relax types to the respective
+       * strings for PETSc.
+       */
+      std::string
+      enum_to_string(RelaxationType relaxation_type);
```

Comment:
Move this function to a `private` section of the class.

Reply:
Also, if you think that one might at some point introduce more parameters one of which is another `enum` type, then maybe calling the function `relaxation_type_to_string` would be a better name.

Reply:
As @nfehn mentioned this could be done implicitly through the `enum` type of the function. But I do not feel strongly here so if you think we should be explicit here I can change it.

Reply:
Also I cannot call the function `enum_to_string` in `PreconditionBoomerAMG::initialize` if it is private. Would you put it in `PreconditionBoomerAMG` instead of `PreconditionBoomerAMG::AdditionalData`?

Reply:
I see. Maybe just call the function `to_string` then; there is already `std::to_string()` and we use that name in a number of places as well.

Regarding where the function should live: I just don't think it should be part of the public interface of the class. You only need it in the `.cc` file, so maybe make it a non-member function in an anonymous namespace in the `.cc` file?

Reply:
Good idea, now it looks really clean! I renamed it to `to_string` and made it a `private` function of `PreconditionBoomerAMG`.

Reply:
@bergbauer You could take a look at the following code as reference:

https://github.com/dealii/dealii/blob/d002b101fcb94246913c664b737039aae4c2ff32/source/fe/mapping_q_cache.cc#L275-L289

Reply:
Thanks for the hint @peterrum ! I pushed a new version with an anonymous namespace in the `.cc` file.